src/nile/signer.py

```diff
@@ -44,10 +44,10 @@ def sign_declare(self, sender, contract_class, nonce, max_fee):
 
         return self.sign(message_hash=transaction_hash)
 
-    def sign_transaction(
+    def sign_invoke_tx(
```

Comment:
I think `sing_transaction` is too generic in this case because we have multiple transactions types, and this method is signing invoke ones specifically (we have a `sign_declare_tx` in the same file, and we probably will add a `sign_deploy_account_tx` for the deployment of accounts from recent protocol updates). 

The invoke name comes from starknet naming, as you can see [here](https://github.com/starkware-libs/cairo-lang/blob/f951a669e264796d53bf7a3cfd23d5bc1f6e54b8/src/starkware/starknet/cli/starknet_cli.py#L55).

If we let `sign_transaction`, why shouldn't this method be used to sign declare transactions? or deploy_account transactions?

Reply:
agreed. then what do you think about renaming `sign_invoke_tx` -> `sign_execute` and dropping `_tx` from them all?

Reply:
+1 to removing the `_tx`